### DISCUSSION TITLE
Top level ErrorBoundary to catch auth errors

### DISCUSSION THREAD

**Discussion by itmecho** [solidjs/solid#973] [Q&A] [answered]
I'm trying to create a top level `ErrorBoundary` which checks for a specific error and triggers the logout process, redirecting the user back to the login page. I can't seem to get it to work. I have created this wrapper component:

```tsx
export const AuthErrorBoundary: Component = (props) => {
  const { logout } = useAuth();
  const c = children(() => props.children);
  return (
    <ErrorBoundary
      fallback={(err: Error) => {
        console.log('here'); // Never gets logged
        if (err instanceof Unauthenticated) {
          logout();
          return <Navigate href="/" />;
        }
      }}
    >
      {c()}
    </ErrorBoundary>
  );
};

```

I've wrapped the component with the error boundary but it doesn't seem to catch the error?

```tsx 
          <Route
            path="/dashboard"
            element={
              <AuthErrorBoundary>
                <DashboardPage />
              </AuthErrorBoundary>
            }
          />
```

In the dashboard page I am using `createResource` and if I check the error there, I can see that it's the correct error type:
```tsx
  const [lists] = createResource(fetchLists, { initialValue: [] });
  createEffect(() => {
    if (lists.error) {
      console.log(lists.error instanceof Unauthenticated); // true
      throw lists.error;
    }
  });

```

I can see the uncaught error in the console as well so it's just like the ErrorBoundary is completely ignoring it? The console.log in the `AuthErrorBoundary`'s fallback method never gets logged. Am I just going about this entirely wrong?

**Comment by itmecho** [comment]
I've just tried directly wrapping the `DashboardPage` component and it now works! Is there something I'm missing about error boundaries and errors bubbling ups?

```tsx
          <Route
            path="/dashboard"
            element={
              <ErrorBoundary
                fallback={(err: Error) => {
                  console.log('here');
                  if (err instanceof Unauthenticated) {
                    return <Navigate href="/" />; // This works!
                  }
                }}
              >
                <DashboardPage />
              </ErrorBoundary>
            }
          />
```

**Comment by ryansolid** [comment, marked as answer]
I see the problem. But its a bit awkward. `children` helper eagerly evaluates the children outside of the error boundary. So they don't end up under the boundary. When you do it directly that doesn't happen.

**Reply by itmecho** [reply]
Ah brilliant, Thank you! That's working as expected now
